src/Configuration/Configuration.php

```diff
@@ -36,12 +40,64 @@ public function bind(Configuration $parent): void
 
     /**
      * @param mixed $value
+     * @param callable $callback
      */
-    public function set(string $name, $value): void
+    public function set(string $name, $value, callable $callback = null): void
     {
+        if (is_callable($callback)) {
+            $this->validations[$name] = $callback;
+        }
+        $callingFile = pathinfo(debug_backtrace()[1]['file'], PATHINFO_FILENAME);
```

Comment:
Unfortunately, this hack should not be used here, this is really not a solution.
I will try to cherry-pick some commits and implement it myself.

Reply:
I don't like it either. It was more to understand better your idea on this.

The content validation is working way better, but I got stuck on this key validation. :(